src/main.rs

```diff
@@ -26,9 +27,16 @@ fn main() {
                 }
             }
 
+            let repository_root = std::env::args().nth(1).unwrap_or_else(|| ".".to_string());
+            let repository_path = PathBuf::from(&repository_root);
+            if !repository_path.as_path().exists() {
```

Comment:
I think something like this is better:

```rust
if repository_path.exists() {
    // ask, construct message and commit
} else {
    eprintln!("Invalid path to repo...");
}
```

Sadly `std::process::exit` doesn't always unwind, so we have to use the program flow to ensure unwinding here.

Reply:
Is there something to unwind? We could avoid the `panic!` by having `main` return a `Result<(), GcmError>` or something similar.

Reply:
No, not now as far as I can see. But it would mean that if something is introduced later on, we wouldn't have to take care of it :)

Reply:
Straight up using a `return` in the `main` function would cause proper unwinding though, right?

Reply:
Not sure to be honest :)